sway-core/src/semantic_analysis/namespace/module.rs

```diff
@@ -27,7 +27,7 @@ pub struct Module {
     /// some library dependency that we include as a submodule.
     ///
     /// Note that we *require* this map to be ordered to produce deterministic codegen results.
```

Comment:
I'm just going to point to this comment here so that we don't merge this by mistake.

We may be able to relax this requirement if the codegen evaluation order no longer relies on the container ordering, but if this cannot be ensured, we should very much not do this.